database/sql/V048__create_settlement_rewards.sql

```diff
@@ -30,7 +33,7 @@ CREATE TABLE auction_prices (
 
 CREATE TABLE auction_participants (
  -- This links to the `auctions` table
- auction_id bigint PRIMARY KEY,
- -- All solvers who submitted a valid solution to the auction.
- participants bytea[]
+ auction_id bigint NOT NULL,
+ -- Solver who submitted a valid solution to the auction.
+ participant bytea NOT NULL
 );
```

Comment:
This table has no  primary key but you are running queries on it of the form `SELECT * FROM auction_participants WHERE auction_id = $1`, which indicates you want an index on at least `auction_id`. Since it doesn't make sense for this table to contain the same `(auction_id, participant)` pair twice, we should probably make `(auction_id, participant) ` the primary key.

Reply:
Fixed in https://github.com/cowprotocol/services/pull/1270

Reply:
How are we supposed to determine the winning participant from this now? Used to the the last entry in the list and we still need to know this info. Specifically for the scenario when the auction is not settled and we want to know who to penalize.

Reply:
In another comment somewhere here we changed the type from an sql array to store one participant per row. This change removed all ordering of participants per auction. This is unrelated to my comment here. Postgres doesn't guarantee any order if there is no ordering constraint in the query.